### ISSUE TITLE
[PROJECT PROPOSAL] User Stories of CNCF Projects integrating Environmental Sustainability

### ISSUE THREAD

**Issue by vallss** [cncf/tag-env-sustainability#361]
### Description
The idea is to gather user stories and collect data on carbon savings from the CNCF ecosystem, the various TAG community members, End User group, GSF, other TAGs, etc.
This would help to create valuable resources (such as documentation, blog posts and/or white papers) that showcase examples of how sustainability solutions can be achieved, explaining the what, why and how.

### Impact

Sharing such experiences would help to give examples that could be followed or taken as a starting point for the utilization of sustainability tools.

### Scope

**In Scope**: ...

**Out of Scope**: ...

### Deliverables

- *Blogs*
- *Documentation*
- [Other] 

### Additional info and comments

- Reference to supporting material:
    - https://github.com/cncf/tag-env-sustainability/issues/324

## TODO

- [ ] Project proposal / suggestion details added to this issue (all section above filled out)
- [ ] Project Leads found (one or more)
  - Project Leads: *link project leads here*
- [ ] Project proposal discussed in the TAG
  - Discussed in the TAG ENV meeting on `XX-XX`
- [ ] Project proposal accepted / rejected by the TAG leads
  - Proposal: `accepted` / `rejected`
- [ ] Project sponsor found (TAG Chair / TAG TL)
- [ ] Project tracking issue opened

cc @cncf/tag-env-sustainability

**Comment by mkorbi** [comment]
a fantastic proposal @vallss, we will need to sharpen the scope and direction of it.
talking about end user we might can collaborate also with the TAB (Technical Advisory Board) that is a group of representatives of end users.

**Comment by vallss** [comment]
thank you so much @mkorbi! Yes, I should be able to add other information in the next few days, hopefully. I also shared the content of this project proposal with @guidemetothemoon and @nikimanoledaki and I think with their support we'll get to better define the missing points and/or complete what I already wrote.

Also, I agree that this could be a great opportunity to collaborate with the TAB!

**Comment by leonardpahlke** [comment]
cc @saiyam1814

**Comment by saiyam1814** [comment]
This is interesting, I got a few questions:
We are already measuring Carbon footprint of via the green reviews project, how is this different form it? 
We are already writing the Kubernetes best practices white paper we can follow the same for others. 
I like the end users part where we can have the case studies blog form the end user companies on what they are doing in their sustainability efforts. 
Overall I think this and Green WG will have some portion in common because if you work on CNCF projects and their Green emissions then I think the white papers should also fall under that for the group(Well that is just my opinion).
End users is something which is really interesting here, but do we need a separate project for that? or we define more goals for this project.

**Comment by nikimanoledaki** [comment]
Hi @saiyam1814, this is different from the Green Reviews WG or the Kubernetes best practices because the focus is on fielding examples from End Users who are actively implementing these tools and patterns in dev/prod environments. The first goal is to take the theory (from the TAG ENV projects mentioned) and showcase concrete examples in practice. The second goal is to gather actual data on energy/carbon optimisations from folks deploying these tools and patterns in their cloud environments.

**Comment by saiyam1814** [comment]
Yes end user stories, usecases and then forming best practices from them sounds good scope for this!!

**Comment by vallss** [comment]
Yes, what made me want to lead this project was the fact that it's very common that end users (I can see sometimes even with the company I work for) don't know where to start or even what to look for in order to include tools that can help their work and their product to be more sustainable and aware of their environmental impact. And the result is that they don't do anything.

**Comment by guidemetothemoon** [comment]
@vallss @leonardpahlke @mkorbi @nikimanoledaki after a discussion with @vallss we worked further on the details around this project proposal and the issue description is now updated with more details. Please take a look and let us know if you have any further input😊 Thanks!

**Comment by leonardpahlke** [comment]
👍 great! We should use the term “adopters” in this proposal which is used in the community to describe companies that use cloud native technology and do not sell it. End users are companies that are part of the CNCF (sponsors).

> In Scope
> * […] Share collected data by creating content on the TAG ENV channels that will be available for the community. […]

We may want to lean on people that we reach out to / approach us with user stories.

> Deliverables
> * Case studies or white papers (potentially, TBD)

I would cut this. Either we plan to write a white paper or not. 

> Deliverables
> * Blog posts

I would clarify: per user story. right? we could also do it differently. What is the goal with each blog?

> Deliverables
> * Gather collected stories in a dedicated section on the TAG ENV website

Where? In the “About" section?

> Goals

We may want to add smth like “Advocating to each adopter to submit to speak about their experiences at a future KCCN conference”

I noticed the Scope/Out of Scope and Goals/Non-Goals in our proposal are quite similar. To clarify, can we revise them to distinctly outline Goals as our target outcomes, Scope as the activities we'll undertake, Non-Goals as what we're intentionally not pursuing, and Out of Scope as activities beyond our current focus? This could help avoid overlap and make our objectives clearer. Thanks!

**Comment by guidemetothemoon** [comment]
@leonardpahlke, thanks for your suggestions. I tried to update it further based on your comments.

**Comment by leonardpahlke** [comment]
🙌 LGTM, I would be good to have a lead with this project! — @saiyam1814 would you be interested?

**Comment by vallss** [comment]
@leonardpahlke we also planned to talk about this project proposal in more depth during next week's TAG general meeting so hopefully we can find a second lead

**Comment by WBurggraaf** [comment]
Hello everyone,

I'm excited to share that on May 6th, I'll be kicking off a series of brainstorming sessions with an associate professor from TU/Delft. We'll be exploring the intersection of AI with green software and cloud-native/open-source solutions for a User Story/BLOG Post. Once I gather enough information, I'll return here to share a rough overview of the story.

Looking forward to updating you all soon!

**Comment by vallss** [comment]
thanks for the update @leonardpahlke. I hope we find a second lead and start this project for real since so many people expressed their interest in supporting it. In case we don't find a second lead, there's a chance that we "postpone" this project for another time this year?

**Comment by WBurggraaf** [comment]
user story:
A team of scientists and researchers is deeply engaged in managing complex machine learning workflows using Kubernetes and Docker. Their primary focus extends beyond traditional ML tasks to optimizing energy consumption and sustainability metrics during model training.

Impact
By showcasing how Kubernetes and Docker can be effectively utilized for managing complex machine learning workflows while optimizing energy consumption and sustainability metrics, the team provides tangible examples of practical applications. This demonstration can inspire and motivate other projects and organizations to adopt similar technologies and practices for their own workflows.

Scope
In Scope
> Reach out to relevant adopters
Done!

> Guide and assist respective adopters throughout the process of creating a user story;
This can take some months because based on feedback it could change to fit better with CNCF and the story is alive and still worked on.


Out of Scope
> Research or advisory on the adoption and implementation of practices, tools, and patterns for cloud-native environmental sustainability.

This technology is being applied in a real-world scenario where the team has been engaged by a company for a specific project. Importantly, the outcomes and findings from this project will be shared openly and publicly with the broader community.

Furthermore, I am spearheading the implementation of OpenTelemetry and advanced measurement practices at my current customer, where I hold the role of solution architect for a Nature Datank containing hundreds of millions of nature observations/measurements. This initiative holds great promise for generating another compelling success story within the next 4-6 months. Specifically, we are focusing on measuring Apache Spark and Postgres Citus sharded databases running on k8s with hundreds of millions of records (we are close to 1 billion internally), which will be made publicly available for all Dutch citizens to download by law.

Short note, gathering data, and creating such a blog will take between 2 to 6 months.

**Comment by lcaywood2** [comment]
I'm interested in helping with this!

**Comment by leonardpahlke** [comment]
@vallss where are we at? Could we walk over this in the next WG Comms meeting (if u have time)

**Comment by vallss** [comment]
We should definitely walk over this in a meeting! Since next week @WBurggraaf will be on holiday and I won't be available on 9th and 10th, I created a lettuce meet to schedule a meeting on week from 15th to 19th. For everyone interested, please select when you're available those days [here](https://lettucemeet.com/l/oJJaD)!

**Comment by leonardpahlke** [comment]
We just talked about the project. The recording is uploaded to [YT](https://youtu.be/91qO6_52v3c).

- We will discuss starting this project tomorrow (July 16). So there is a place for everyone to provide feedback.
- We would like to limit this project time wise until end of the year. In the new year we can assess the project and, depending on discussions and project results, that may integrate the effort into WG Comms (cc @claire-fletcher). There is not a real limit to how many user stories are interesting to discuss, so this could transform next year into a continuous effort if we like. 
- Since this project evolves around user stories, we discussed the option that Project leads are leads of one of the (or more) user stories. Person A is the lead of user story 1, Person B is the lead of user story 2, ...
- To give some more guidance, it would be helpful to create a user stories template or examples of user stories that can serve as an example. 
- It may be the case that companies have a specific understanding of what sustainability means to them, and that is great and interesting. It would be good to put this into context. If one user story is more focused on the FinOps overlap, or more about observability, it would be valuable to highlight this in some kind of discussion after the user story, so everyone can put this user story into context. User story A is a great example of how Company X tackled energy observability of the network. 
- @guidemetothemoon or @leonardpahlke will host the project meets, which will be scheduled soon every two weeks. It is possible to organize additional deep dives for each user story. 

@guidemetothemoon @WBurggraaf please add to this if i missed something 😄

**Comment by vallss** [comment]
I agree with all the points of @leonardpahlke's comment above. I really appreciate the idea of having multiple leads and each one "responsible" of one user story. I think this would also help all those cases in which one single person maybe has more people to contact because we could share between other leads that have less contacts and are interested in helping with this. 

About this @WBurggraaf, I can help you with the multiple people/companies you mentioned during the meeting!

I also agree that before starting a user story, we clarify what the people/company has to share and what sustainability means to them.
